crates/sys/src/lib.rs

```diff
@@ -48,21 +48,79 @@ mod bindings;
 /// ```
 /// flipperzero_sys::crash!("Hello world!");
 /// ```
+///
+/// Crash the system with default *"Fatal Error"* message:
+///
+/// ```
+/// flipperzero_sys::crash!();
+/// ```
 #[macro_export]
 macro_rules! crash {
-    ($msg:expr $(,)?) => {{
-        const MESSAGE: *const ::core::primitive::i8 =
-            match ::core::ffi::CStr::from_bytes_with_nul(
-                ::core::concat!($msg, "\0").as_bytes(),
-            ) {
-                Ok(cstr) => cstr.as_ptr(),
-                Err(error) => panic!("message contains NULs"),
+    ($($msg:expr)? $(,)?) => {{
+        // Optional message
+        let mut message: *const i8 = ::core::ptr::null();
+        $(
+            message = match ::core::ffi::CStr::from_bytes_with_nul(::core::concat!($msg, "\0").as_bytes()) {
+                Err(_) => c"nul in crash message".as_ptr(),
+                Ok(msg) => msg.as_ptr(),
             };
+        )?
+
+        unsafe {
+            // Crash message is passed via r12
+            ::core::arch::asm!(
+                "ldr pc,=__furi_crash_implementation",
+                in("r12") message,
+                options(nomem, nostack))
+            ;
```

Comment:
```suggestion
                options(nomem, nostack),
            );
```

Feels  like a more rustfmt'ish  formatting

Reply:
That's odd. I agree. I wonder if rustfmt did something weird on me or if it was just me. 😅